hybrid/compartment_examples/inter_comp_call/secure-redirect_clr/Makefile.morello-hybrid

```diff
@@ -0,0 +1,10 @@
+# Copyright (c) 2021 The CapableVMs "CHERI Examples" Contributors.
+# SPDX-License-Identifier: MIT OR Apache-2.0
+
+SHARED_SOURCES := shared.s
+CFILES := $(wildcard *.c)
```

Comment:
I don't think these are correct. Using `Makefile.simple`:

- Each entry in `CFILES` maps onto a binary target with the same name (minus `.c`). Defaults to `*.c`.
- Each entry in `HFILES` is treated as a dependency, but not otherwise used. Defaults to `include/*.h`.
- Each entry in `SHARED_SOURCES` is treated as a dependency and passed to the compile command for _every_ binary target (as per `CFILES`).

I could, perhaps, have chosen better names, but in my defence, they worked for all the examples at the time.

To make this work as the `secure-redirect_clr` source files are written currently (reusing `Makefile.simple`):

```
SHARED_SOURCES := shared.S    # Note capital .S.
# CFILES := $(wildcard *.c)   # This is the default anyway. You can specify main.c if you prefer.
HFILES := $(wildcard *.h) $(wildcard ../../../include/*.h) $(wildcard ../../../../include/*.h) $(wildcard *.s)
    # ... or whatever else you used. Note that *.s is listed here because they're #included in shared.S.
```

This is a bit confusing because you have some `.s` files that are `#include`d, so they have to be included under `HFILES`, not `SHARED_SOURCES`.

For me, that causes `make` to try the following (which I think is correct):
```
[...]/clang --config cheribsd-morello-hybrid.cfg  main.c shared.S     -o bin/morello-hybrid/main
```

It doesn't build like that, because I lack `compartments.s`.

----

Whilst that would work, it's probably not ideal. It would be better to put all the .s (and .S) files into `SHARED_SOURCES` and drop the `#include`s. You'd have to declare the functions if you want to call them (like `.global switch_compartment`) but then everything works more conventionally, which is useful in an example!

I'd aim for this, and make the source files match:

```
SHARED_SOURCES := $(wildcard *.S)   # Just make all the files .S to keep this simple.
# CFILES := [default]
HFILES := $(wildcard include/*.h) $(wildcard ../../../include/*.h) $(wildcard ../../../../include/*.h)
     # Assuming that `main.h` is moved under `include`, like most other examples.
```

... so Make tries:

```
#                                     CFILES[0] ---v    /----------------- SHARED_SOURCES ----------------\
[...]/clang --config cheribsd-morello-hybrid.cfg main.c shared.S compartments.S compartments-redirect_clr.S -o bin/morello-hybrid/main
```

Reply:
@0152la Has this comment been dealt with?

Reply:
Not yet (that's why I changed it to a draft PR).

Reply:
Ah!

Reply:
This should be now fixed in https://github.com/capablevms/cheri-examples/pull/57/commits/44440e75a75af8528307a00193052f3155aa9bed. (with the corresponding changes for the second example done in https://github.com/capablevms/cheri-examples/pull/57/commits/4c9279a8a5e169b274b030a229d3f563f457043d)